arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/BaseVectorizedParquetValuesReader.java

```diff
@@ -80,17 +80,23 @@ public BaseVectorizedParquetValuesReader(int maxDefLevel, boolean setValidityVec
     this.setArrowValidityVector = setValidityVector;
   }
 
-  public BaseVectorizedParquetValuesReader(
-      int bitWidth,
-      int maxDefLevel,
-      boolean setValidityVector) {
+  public BaseVectorizedParquetValuesReader(int bitWidth, int maxDefLevel, boolean setValidityVector) {
     this.fixedWidth = true;
     this.readLength = bitWidth != 0;
     this.maxDefLevel = maxDefLevel;
     this.setArrowValidityVector = setValidityVector;
     init(bitWidth);
   }
 
+  public BaseVectorizedParquetValuesReader(int bitWidth, int maxDefLevel, boolean readLength,
+                                           boolean setValidityVector) {
+    this.fixedWidth = true;
+    this.readLength = readLength;
```

Comment:
It seems a little strange to me that we have this constructor which we only use when readLength is false. Perhaps we should swap the original constructor's code to call this constructor?

```java
  public BaseVectorizedParquetValuesReader(int bitWidth, int maxDefLevel, boolean setValidityVector) {
    this(bitWidth, maxDefLevel, bitWidth != 0, setValidityVector)
  }
```

Reply:
Thanks! Good suggestion.